src/components/NNudge/NNudge.vue

```diff
@@ -0,0 +1,323 @@
+  
+      </div>
+  
+    </div>
+  </template>
+    
+  <script>
+  import NHeader from '../NHeading/index';
+  import NText from '../NText';
+  import NIcon from '../NIcon/NIcon.vue';
+  
+  export default {
+    name: 'Nudge',
+    components: {
+      NHeader,
+      NText,
+      'nitrozen-icon': NIcon,
+    },
+    props: {
+        showNudge: {
+        type: Boolean,
+        default: false,
+      },
+      title: {
+        type: String,
+        default: '',
+      },
+      description: {
+        type: String,
+        default: '',
+      },
+      nudgeType: {
+        type: String,
+        default: '', // Default type (info, success, warning, error, action, etc.)
+      },
+      icon: {
+        type: String,
+        default: null,
+      },
+      showPrimaryButton: {
+        type: Boolean,
+        default: false,
+      },
+      primaryButtonLabel: {
+        type: String,
+        default: 'Primary',
+      },
+      showSecondaryButton: {
+        type: Boolean,
+        default: false,
+      },
+      secondaryButtonLabel: {
+        type: String,
+        default: 'Secondary',
+      },
+    },
+    data() {
+      return {
+        nudgeKey: 0,
+      };
+    },
+    methods: {
+      showToastMessage() {
+        if (this.showNudge) {
+          this.nudgeKey++;
+  
+          if (this.nudgeType !== 'action') {
+            setTimeout(() => {
+              this.hideToast();
+            }, 5000);
+          }
+        }
+      },
+      handleCancelClick() {
+        this.showNudge = false;
+      },
+      hideToast() {
+        this.$emit('onClose');
+      },
+      handleClick() {
+        this.$emit('onClose');
+      },
+      handlePrimaryClick() {
+        this.$emit('onClickPrimary');
+      },
+      handleSecondaryClick() {
+        this.$emit('onClickSecondary');
+      },
+    },
+    watch: {
+    showNudge: {
+        immediate: true,
+        handler(newVal) {
+          if (newVal && this.nudgeType !== 'action') {
+            setTimeout(() => {
+              this.$emit('onClose');
+            }, 5000);
+          }
+        },
+      },
+    },
+  };
+  </script>
+    
+  <style>
+  .nudge{
+  
+  visibility: visible;
+  
+  min-width: 360px;
+  
+  margin-right: 24px;
+  
+  margin-top: 24px;
+  
+  background-color: rgba(0, 0, 0, .65);
+  
+  color: #fff;
+  
+  text-align: center;
+  
+  border-radius: 16px;
+  
+  padding: 16px;
+  
+  position: fixed;
+  
+  z-index: 70;
+  
+  font-size: 12px;
+  
+  letter-spacing: -0.06px;
+  top: 0;
+  right: 0;
+  
+  
+  @media only screen and (max-width: 767px)  {
+  
+  left:50%;
+  
+  margin-left:-180px;
+  
+  }
+  
+
+  
+  }
+  
+  
+  
+  
+  @keyframes tfadein {
+  
+  from {left: 0; opacity: 0;}
+  
+  to {left: 10; opacity: 1;}
+  
+  }
+  
+  
+  
+  
+  @keyframes tfadeout {
+  
+  from {left: 10%; opacity: 1;}
+  
+  to {left: 0; opacity: 0;}
+  
+  }
+  
+  
+  
+  
+  .nudge.info {
+  
+    background-color: #007bff;
```

Comment:
can we use colour Variables here instead of hardcoded ones
mention colour variable.less and import this for all above and below ones